app/components/userFilter/userFilterDirective.js

```diff
@@ -18,17 +18,24 @@
 
             function UserFilterLink(scope) {
 
-                scope.userSearch = userSearch;
+                scope.userfilterInput = '';
 
-                scope.showUserFilterInfo = showUserFilterInfo;
+                scope.userSearch = userSearch;
 
-                scope.userfilterInput = '';
+                scope.resetUser = resetUser;
 
+                scope.showUserFilterInfo = showUserFilterInfo;
                 /**
                  *
                  */
                 function userSearch() {
-                    HeatMapSourceGenerator.search(scope.userfilterInput);
+                    HeatMapSourceGenerator.filterObj.setUser(scope.userfilterInput);
```

Comment:
please don't set the filterObj like this. use a method for it.
I realised this as well and added a `searchUser` method to `HeatMapSourceGenerator`